### ISSUE TITLE
Detect and reject an "argument constraint factory method" that produces two constraints

### ISSUE THREAD

**Issue by blairconrad** [FakeItEasy/FakeItEasy#1628]
It's a weird situation to be in, but if it comes up, we should complain about it.

Specifically, this spec fails, because no exception is thrown (we just use the first constraint, not even the one that's returned from the method):

```c#
[Scenario]
public static void ConstraintFactoryThatMakesTwoConstraints(
    IHaveAnObjectParameter fake,
    Func<object> constraintFactory,
    Exception exception)
{
    "Given a fake"
        .x(() => fake = A.Fake<IHaveAnObjectParameter>());

    "And a delegate that produces two constraints"
        .x(() => constraintFactory = () =>
            {
                A<object>.That.Matches(i => i is object);
                return A<object>.That.Matches(i => i is null);
            });

    "When I try to configure a method of the fake with this delegate"
        .x(() => exception = Record.Exception(() => A.CallTo(() => fake.Bar(constraintFactory())).Returns(1)));

    "Then the call configuration throws a FakeConfigurationException"
        .x(() => exception.Should().BeAnExceptionOfType<FakeConfigurationException>());
}
```

(Of course, if we weren't using a method to create constraints, and just specifying one inline via an expression, it's only possible to create a single constraint.)

**Comment by blairconrad** [comment]
Found while I was investigating nullability warnings, and (re)trying to understand the argument constraint capturing.
@thomaslevesque, assuming you've no objection to fixing this, I'd like to take it.

**Comment by thomaslevesque** [comment]
Interesting. That's definitely a corner case, but we should fix it nevertheless.

> (Of course, if we weren't using a method to create constraints, and just specifying one inline via an expression, it's only possible to create a single constraint.)

Are you sure? What about `A.CallTo(() => fake.Bar(A<int>.That.IsEven() + A<int>.That.IsOdd()))`?

(`IsOdd` and `IsEven` don't actually exist, it's just to illustrate)

**Comment by blairconrad** [comment]
`A.CallTo(() => fake.Bar(A<int>.That.IsEven() + A<int>.That.IsOdd()))` throws `System.InvalidOperationException : An argument constraint, such as That, Ignored, or _, cannot be nested in an argument.`.
`ExpressionArgumentConstraintFactory.CheckArgumentExpressionIsValid` does a pretty good job when it looks at the argument constraint expression!

(As an aside, maybe we should be throwing a `FakeConfigurationException`.)

**Comment by thomaslevesque** [comment]
Good point!

> (As an aside, maybe we should be throwing a `FakeConfigurationException`.)

Maybe. I don't think we're very consistent in the types of exceptions we throw...

**Comment by blairconrad** [comment]
This change has been released as part of [FakeItEasy 5.3.0](https://github.com/FakeItEasy/FakeItEasy/releases/tag/5.3.0).
